Update Makefile to supporte latest dcos-e2e/minidcos

Fabs opened the pull request:
Worth of notice:
- Renamed `dcos-e2e` command to `minidcos` to work with the newest version of `minidcos`.
- Made a change on `dcos-docker-web` target runs even if there is no `socat` container.
- Renamed some filters to get rid of the `dcos-e2e` string.

urbanserj commented:
@Fabs could you please rename `dcos-docker-*` goals and variables to `mini-dcos-*`?

Fabs commented:
- Extended the refactor one step further to include the target/goals and the variables. I changed it to `MINIDCOS_` and `minidcos-` because the bash command is `minidcos` and not `mini-dcos`. @urbanserj what you think?
- Fixed the documentation
- Fixed a label on `mini-dcos-destroy`

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
+	    --filter label=minidcos-web-port=$(MINIDCOS_WEB_PORT) \
 	    --format '{{.ID}}' \
-	| xargs docker kill > /dev/null
+	| xargs --no-run-if-empty docker kill > /dev/null
```
It doesn't work on macos.

```
usage: xargs [-0opt] [-E eofstr] [-I replstr [-R replacements]] [-J replstr]
             [-L number] [-n number [-x]] [-P maxprocs] [-s size]
             [utility [argument ...]]
```

Here is my suggestion:

```suggestion
	| while read id; do [ -z "$id" ] || docker kill $id; done > /dev/null
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
-	    --filter label=dcos-e2e-web-id=$(DCOS_DOCKER_CLUSTER_ID) \
-	    --filter label=dcos-e2e-web-port=$(DCOS_DOCKER_WEB_PORT) \
+minidcos-web: minidcos-create
+	-@ docker ps \
```
Please let's make it posix compatible and don't ignore the exit code.

Fabs replied:
```suggestion
	@ docker ps \
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
 	    --format '{{.ID}}' \
-	| xargs docker kill > /dev/null
-	@ dcos-docker destroy --cluster-id $(DCOS_DOCKER_CLUSTER_ID)
+	| xargs --no-run-if-empty docker kill > /dev/null
```
```suggestion
	| while read id; do [ -z "$id" ] || docker kill $id; done > /dev/null
```

Fabs commented on README.md:
```diff
@@ -52,8 +52,7 @@ You can build, check, and test dcos-net in a development image using
 
-To check your dcos-net build on DC/OS you can use DC/OS E2E a.k.a.
-[dcos-docker CLI](http://dcos-e2e.readthedocs.io/en/latest/cli.html). In order to do so please repeat the following steps:
+To check your dcos-net build on DC/OS you can use [minidcos](https://dcos-e2e-cli.readthedocs.io/en/latest/). In order to do so please repeat the following steps:
```
```suggestion
To check your dcos-net build on DC/OS you can use [miniDC/OS](https://dcos-e2e-cli.readthedocs.io/en/latest/). In order to do so please repeat the following steps:
```

Fabs commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
 
 ##
-## DC/OS E2E
+## MINI DC/OS
```
```suggestion
## miniDC/OS
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
 	@ docker ps \
-	    --filter label=dcos-e2e-web-id=$(DCOS_DOCKER_CLUSTER_ID) \
-	    --filter label=dcos-e2e-web-port=$(DCOS_DOCKER_WEB_PORT) \
+	    --filter label=minidcos-web-id=$(MINIDCOS_CLUSTER_ID) \
```
As of miniDC/OS release, the labels have changed, but they don't have `minidcos` prefixes. It looks like `dcos-e2e.cluster_id` and `dcos-e2e.web-port` are what we should use now.

```
$ docker ps --format '{{.ID}}' | xargs docker inspect | jq '.[] | .Config.Labels + {name: .Name}'        
{
  "dcos_e2e.cluster_id": "default",
  "dcos_e2e.node_type": "agent",
  "dcos_e2e.variant": "oss",
  "dcos_e2e.workspace_dir": "/var/folders/70/8cktdr690m7_n95n__x9r2p80000gp/T/a81214c5050845f5b7617a14a87a68dd",
  "org.label-schema.build-date": "20181205",
  "org.label-schema.license": "GPLv2",
  "org.label-schema.name": "CentOS Base Image",
  "org.label-schema.schema-version": "1.0",
  "org.label-schema.vendor": "CentOS",
  "name": "/dcos-e2e-default-9890a-agent-0"
}
{
  "dcos_e2e.cluster_id": "default",
  "dcos_e2e.node_type": "master",
  "dcos_e2e.variant": "oss",
  "dcos_e2e.workspace_dir": "/var/folders/70/8cktdr690m7_n95n__x9r2p80000gp/T/a81214c5050845f5b7617a14a87a68dd",
  "org.label-schema.build-date": "20181205",
  "org.label-schema.license": "GPLv2",
  "org.label-schema.name": "CentOS Base Image",
  "org.label-schema.schema-version": "1.0",
  "org.label-schema.vendor": "CentOS",
  "name": "/dcos-e2e-default-9890a-master-0"
}
```

Fabs replied:
I do not understand why you picked `dcos-e2e.cluster_id` (with a hyphen on dcos-e2e) instead of `dcos_e2e.cluster_id`(with underscore) as we see on the json above.

urbanserj replied:
Oops, you are right. Here are the correct label names: `dcos_e2e.cluster_id` and `dcos_e2e.web_port`.

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
+	    --filter label=minidcos-web-port=$(MINIDCOS_WEB_PORT) \
 	    --format '{{.ID}}' \
-	| xargs docker kill > /dev/null
+	| while read id; do [ -z "$id" ] || docker kill $id; done > /dev/null
```
```
docker ps --filter label=dcos-e2e.cluster_id=$(MINIDCOS_CLUSTER_ID) --filter label=dcos-e2e.web-port=$(MINIDCOS_WEB_PORT) … |
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
-	    alpine/socat TCP4-LISTEN:$(DCOS_DOCKER_WEB_PORT),bind=0.0.0.0,reuseaddr,fork,su=nobody \
-	                 TCP4:$(shell dcos-docker inspect --cluster-id $(DCOS_DOCKER_CLUSTER_ID) | \
+	                   sed -re 's/-master-0/-web-$(MINIDCOS_WEB_PORT)/') \
+	    --label minidcos-web-id=$(MINIDCOS_CLUSTER_ID) \
```
```
--label dcos-e2e.cluster_id=$(MINIDCOS_CLUSTER_ID)
--filter dcos-e2e.web-port=$(MINIDCOS_WEB_PORT)
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,76 @@ dev: dev-install dev-stop
 
-dcos-docker-destroy:
+minidcos-destroy:
 	@ docker ps \
```
```
docker ps --filter label=dcos-e2e.cluster_id=$(MINIDCOS_CLUSTER_ID) --filter label=dcos-e2e.web-port … |
```

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,77 @@ dev: dev-install dev-stop
 	    --format '{{.ID}}' \
-	| xargs docker kill > /dev/null
-	@ dcos-docker destroy --cluster-id $(DCOS_DOCKER_CLUSTER_ID)
+	    | while read id; do [ -z "$id" ] || docker kill $id; done > /dev/null
```
```suggestion
	| while read id; do [ -z "$$id" ] || docker kill $$id; done > /dev/null
```

`$i` is not defined in `Makefile` and it always tried to kill container `d`. To avoid the issue, we must follow quoting rules. When I was suggesting the command I checked it in shell, but not in Makefile. My bad.

https://www.gnu.org/software/make/manual/html_node/Variables-in-Recipes.html

Variable and function references in recipes have identical syntax and semantics to references elsewhere in the makefile. They also have the same quoting rules: if you want a dollar sign to appear in your recipe, you must double it (‘$$’). For shells like the default shell, that use dollar signs to introduce variables, it’s important to keep clear in your mind whether the variable you want to reference is a make variable (use a single dollar sign) or a shell variable (use two dollar signs).

Fabs replied:
Oh, yeah I remember that I overlooked that too :/. 

I brought my mac to the office today. I was planning to test everything on Mac myself later today. I am glad you are helping me get this ready, but if you think you are wasting too much time with it, I will gladly make sure all runs smoothly this afternoon.

urbanserj commented on Makefile:
```diff
@@ -125,76 +125,77 @@ dev: dev-install dev-stop
+	    --filter label=dcos_e2e.web_port=$(MINIDCOS_WEB_PORT) \
 	    --format '{{.ID}}' \
-	| xargs docker kill > /dev/null
+	    | while read id; do [ -z "$id" ] || docker kill $id; done > /dev/null
```
```suggestion
	| while read id; do [ -z "$$id" ] || docker kill $$id; done > /dev/null
```

Fabs commented on Makefile:
```diff
@@ -125,76 +125,77 @@ dev: dev-install dev-stop
-	    --label dcos-e2e-web-port=$(DCOS_DOCKER_WEB_PORT) \
-	    alpine/socat TCP4-LISTEN:$(DCOS_DOCKER_WEB_PORT),bind=0.0.0.0,reuseaddr,fork,su=nobody \
-	                 TCP4:$(shell dcos-docker inspect --cluster-id $(DCOS_DOCKER_CLUSTER_ID) | \
+	                   sed -re 's/-master-0/-web-$(MINIDCOS_WEB_PORT)/') \
```
on my mac, this only work if I use `gnu-sed`. With `sed` I get:

```
sed: illegal option -- r
```
Are you also using gnu-sed @urbanserj? I saw a thread where folks fixed it, but I was planning just to document that you need gnu-sed on mac. What do you think @urbanserj?

urbanserj replied:
Yes, I have gnu sed installed. Could you please remove `-r`, i.e. `s/-re/-e/`? it should make `Makefile` fully posix compatible.
